### ISSUE TITLE
Usage as an asynchronous context manager

### ISSUE THREAD

**Issue by mivade** [cole/aiosmtplib#40]
It would be nice if the `SMTP` class could be used as an async context manager. From digging around, it looks like it might be possible, but it isn't documented.

**Comment by cole** [comment]
Good point. @mivade is the example here clear to you? https://aiosmtplib.readthedocs.io/en/latest/overview.html#connecting-via-async-context-manager

**Comment by mivade** [comment]
Perfect!
